### PR TITLE
VideoPress: implement filter section UI in the dashboard

### LINKED ISSUE #26144: VideoPress: finish the Filters component implementation
Image

### PR DESCRIPTION
**Pull request by retrofox**
This PR implements the first approach of the VideoPress dashboard page filters section. For this purpose, I had to add the `weight` property to the `<Text />` component to achieve the desired design.

Remaining issues:

* https://github.com/Automattic/jetpack/issues/26167
* https://github.com/Automattic/jetpack/issues/26168

Fixes https://github.com/Automattic/jetpack/issues/26144

#### Changes proposed in this Pull Request:

* VideoPress: implement filter section UI in the dashboard
* Component: add `weight` prop to `<Text />` component

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Go to the VideoPress dashboard
* Confirm when clicking on the Filter button the filters section shows up.

https://user-images.githubusercontent.com/77539/189633682-3ba61645-edfe-477f-840b-3fca316c19c3.mov

### CODE REVIEW

**Review by renatoagds — COMMENTED** [review]
I have a specific point about the `weight` support at `Text`.

This implementation will make us support two different strategies. Currently (and based on Design decision), we use only variants, and we navigate through `weight` using `variants` too, ex: `headline-small` -> `headline-small-regular`.

For me, we should not mix different strategies, since it makes the component not clear for who's using it and could generate strange behaviors (and bugs). Ex:

```javascript
<Text variant="headline-small-regular" /> // This will apply `font-weight: 700` since bold is default (already a bug in the current implementation)
<Text variant="headline-small" weight="regular" /> // This will apply `font-weight: 400`, which is not the expected approach based on Design specs
```

So, if we want to use this approach, we should normalize and have variants unrelated to `weight` and sync that with the Design team. Based on that, I would say to split this change to another PR so we could test and discuss it better.

**Comment by retrofox** [comment]
> This implementation will make us support two different strategies...
> ...we should not mix different strategies, ...

Agree

> So, if we want to use this approach, we should normalize and have variants unrelated to `weight` and sync that with the Design team. Based on that, I would say to split this change to another PR so we could test and discuss it better.

Can we do that in a follow-up PR? Happy to research in a consistent way to deal with the typographies. Also, feel free to define it. I can implement it.

**Comment by renatoagds** [comment]
> Can we do that in a follow-up PR? Happy to research in a consistent way to deal with the typographies. Also, feel free to define it. I can implement it.

Sure, but if you merge these changes, things will break since we have unexpected behaviors as described in past examples, so vote to undo these changes on `Text` and implement it in a follow-up PR where we could discuss and test better across other places.

File: projects/packages/videopress/src/client/admin/components/checkbox/style.module.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .checkbox {
 	&[type="checkbox"] {
 		all: unset;
-		border: 1px solid black;
+		border: 1px solid var(--jp-black);;
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/checkbox/style.module.scss:4]: Thanks 💯

**Review by renatoagds — DISMISSED** [review]
LGTM

**Review by oskosk — APPROVED** [review]

**Pull request merged**